File: app/src/main/java/com/cristhian/gametobeat/ui/login/LoginActivity.java
package com.cristhian.gametobeat.ui.login;

import android.content.Intent;
import android.os.Bundle;
import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import android.view.View;
import android.widget.EditText;
import android.widget.Toast;

import com.cristhian.gametobeat.R;
import com.cristhian.gametobeat.ui.list.GameListActivity;
import com.cristhian.gametobeat.util.LoginUtil;
import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.Task;
import com.google.firebase.auth.AuthResult;
import com.google.firebase.auth.FirebaseAuth;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;

public class LoginActivity extends AppCompatActivity {

    @BindView(R.id.et_email)
    EditText mEmail;

    @BindView(R.id.et_password)
    EditText mPassword;

    private FirebaseAuth mAuth;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login);

        ButterKnife.bind(this);
        mAuth = FirebaseAuth.getInstance();

        if (mAuth.getCurrentUser() != null) {
            Intent intent = new Intent(LoginActivity.this, GameListActivity.class);
            startActivity(intent);
            finish();
        }
    }

    @OnClick(R.id.bt_login)
    public void login() {

        // Reset errors
        mEmail.setError(null);
        mPassword.setError(null);

        String email = mEmail.getText().toString();
        String password = mPassword.getText().toString();

        boolean cancel = false;
        View focusView = null;

        if (!LoginUtil.isPasswordValid(password)) {
            mPassword.setError(getString(R.string.password_invalid));
            cancel = true;
            focusView = mPassword;
        }

        if (!LoginUtil.isEmailValid(email)) {
            mEmail.setError(getString(R.string.email_invalid));
            cancel = true;
            focusView = mEmail;
        }

        if (cancel) {
            focusView.requestFocus();
        } else {

            mAuth.signInWithEmailAndPassword(email, password)
                    .addOnCompleteListener(new OnCompleteListener<AuthResult>() {
                @Override
                public void onComplete(@NonNull Task<AuthResult> task) {
                    if (!task.isSuccessful()) {
                        Toast.makeText(LoginActivity.this, getString(R.string.login_error), Toast.LENGTH_LONG).show();
                    } else {

                        Intent intent = new Intent(LoginActivity.this, GameListActivity.class);
                        startActivity(intent);

                        finish();
                    }
                }
            });
        }
    }

    @OnClick(R.id.tv_forgot_password)
    public void forgotPassword() {
        Intent intent = new Intent(this, ResetPasswordActivity.class);
        startActivity(intent);
    }

    @OnClick(R.id.tv_register_now)
    public void registerNow() {
        Intent intent = new Intent(this, RegisterActivity.class);
        startActivity(intent);
    }
}

File: app/src/main/java/com/cristhian/gametobeat/ui/login/LoginActivity.kt
package com.cristhian.gametobeat.ui.login

import android.content.Intent
import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.cristhian.gametobeat.R
import com.cristhian.gametobeat.ui.list.GameListActivity
import com.cristhian.gametobeat.util.LoginUtil
import com.google.firebase.auth.FirebaseAuth
import kotlinx.android.synthetic.main.activity_login.*

class LoginActivity : AppCompatActivity() {

    lateinit var mFirebaseAuth: FirebaseAuth

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_login)

        configureFirebaseAuth()

        if (::mFirebaseAuth.isInitialized && mFirebaseAuth.currentUser != null) {
            val intent = Intent(this@LoginActivity, GameListActivity::class.java)
            startActivity(intent)
            finish()
        }
        configureOnClicks()
    }

    private fun configureFirebaseAuth() {
        mFirebaseAuth = FirebaseAuth.getInstance()
    }

    private fun configureLogin() {

        // Reset errors
        et_email.error = null
        et_password.error = null

        if (!LoginUtil.isEmailValid(et_email.text.toString())) {
            et_email.error = getString(R.string.email_invalid)
            et_email.requestFocus()
            return
        }

        if (!LoginUtil.isPasswordValid(et_password.text.toString())) {
            et_password.error = getString(R.string.password_invalid)
            et_password.requestFocus()
            return
        }

        //logar
        mFirebaseAuth.signInWithEmailAndPassword(et_email.text.toString(), et_password.text.toString())
                .addOnCompleteListener {
                    if (!it.isSuccessful) {
                        Toast.makeText(
                                this@LoginActivity,
                                getString(R.string.login_error),
                                Toast.LENGTH_LONG
                        ).show()
                    } else {
                        val intent = Intent(this@LoginActivity, GameListActivity::class.java)
                        startActivity(intent)
                        finish()
                    }
                }

    }

    private fun configureOnClicks() {

        bt_login.setOnClickListener {
            configureLogin()
        }

        tv_forgot_password.setOnClickListener {
            val intent = Intent(this, ResetPasswordActivity::class.java)
            startActivity(intent)
        }

        tv_register_now.setOnClickListener {
            val intent = Intent(this, RegisterActivity::class.java)
            startActivity(intent)
        }
    }
}